pkg/snapshot/backup.go

```diff
@@ -0,0 +1,272 @@
+	if err != nil {
+		return errors.Wrap(err, "failed to create clientset")
+	}
+
+	_, err = veleroClient.Backups(appNamespace).Create(veleroBackup)
+	if err != nil {
+		return errors.Wrap(err, "failed to create velero backup")
+	}
+
+	return nil
+}
+
+func ListBackupsForApp(appID string) ([]*types.Backup, error) {
+	appNamespace := os.Getenv("POD_NAMESPACE")
+	if os.Getenv("DEV_NAMESPACE") != "" {
+		appNamespace = os.Getenv("DEV_NAMESPACE")
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cluster config")
+	}
+
+	veleroClient, err := veleroclientv1.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create clientset")
+	}
+
+	veleroBackups, err := veleroClient.Backups(appNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list velero backups")
+	}
+
+	backups := []*types.Backup{}
+
+	for _, veleroBackup := range veleroBackups.Items {
+		backup := types.Backup{
+			Name:   veleroBackup.Name,
+			Status: string(veleroBackup.Status.Phase),
+			AppID:  appID,
+		}
+
+		if veleroBackup.Status.StartTimestamp != nil {
+			backup.StartedAt = &veleroBackup.Status.StartTimestamp.Time
+		}
+		if veleroBackup.Status.CompletionTimestamp != nil {
+			backup.FinishedAt = &veleroBackup.Status.CompletionTimestamp.Time
+		}
+		if veleroBackup.Status.Expiration != nil {
+			backup.ExpiresAt = &veleroBackup.Status.Expiration.Time
+		}
+		sequence, ok := veleroBackup.Annotations["kots.io/app-sequence"]
+		if ok {
+			s, err := strconv.ParseInt(sequence, 10, 64)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to parse app sequence")
+			}
+
+			backup.Sequence = s
+		}
+		if backup.Status == "" {
+			backup.Status = "New"
+		}
+
+		trigger, ok := veleroBackup.Annotations["kots.io/snapshot-trigger"]
+		if ok {
+			backup.Trigger = trigger
+		}
+
+		volumeCount, volumeCountOk := veleroBackup.Annotations["kots.io/snapshot-volume-count"]
+		if volumeCountOk {
+			i, err := strconv.Atoi(volumeCount)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to convert volume-count")
+			}
+			backup.VolumeCount = i
+		}
+
+		volumeSuccessCount, volumeSuccessCountOk := veleroBackup.Annotations["kots.io/snapshot-volume-success-count"]
+		if volumeSuccessCountOk {
+			i, err := strconv.Atoi(volumeSuccessCount)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to convert volume-success-count")
+			}
+			backup.VolumeSuccessCount = i
+		}
+
+		volumeBytes, volumeBytesOk := veleroBackup.Annotations["kots.io/snapshot-volume-bytes"]
+		if volumeBytesOk {
+			i, err := strconv.ParseInt(volumeBytes, 10, 64)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to convert volume-bytes")
+			}
+			backup.VolumeBytes = i
+			backup.VolumeSizeHuman = "Many TB"
+		}
+
+		if backup.Status != "New" && backup.Status != "InProgress" {
+			if !volumeBytesOk || !volumeSuccessCountOk || !volumeBytesOk {
```

Comment:
This needs a comment.  I don't understand this logic.... Why when something is not ok, we update snapshot info, but not otherwise?